Bluetooth: Audio: CCID module

asbjornsabo opened the pull request:
The Content Control ID (CCID) characteristic is a separately defined characteristic used in the media control service and in the telephone bearer service in LE Audio.  The purpose of the CCID is to provide device-unique identifiers for content control services on that device.

This PR is part of the upmerge of le-audio host code from the topic-le-audio branch.  This CCID module has existed (unchanged) in that branch since before March 2020 (and in another repo before that).

asbjornsabo committed:
Bluetooth: Audio: UUID values for the CCID characteristic

UUID values for the Content Control ID characteristic

The Content Control ID characteristic is a separately defined
characteristic used in the media control service and in the telephone
bearer service in LE Audio.

This commit is part of the upmerge of le-audio from the topic-le-audio
branch.

Signed-off-by: Asbjørn Sæbø <asbjorn.sabo@nordicsemi.no>

asbjornsabo committed:
Bluetooth: Audio: Content Control ID module

This commit adds the content control ID (CCID) module from the
topic-le-audio branch.  This is required to the le-audio media control
files and call control files, which depend upon CCID.

The commit is a pure copy of the files and content in the
topic-le-audio branch, with the following exceptions:

- files are in bluetooth/audio instead of bluetooth/host/audio, with
  some include paths updated as a consequence
- as a consequence, CMake files and Kconfig files updates are done in
  other locations

The CCID module is written by
Emil Gydesen <emil.gydesen@nordicsemi.no>

Signed-off-by: Asbjørn Sæbø <asbjorn.sabo@nordicsemi.no>

Thalley reviewed (changes requested):
The placement of the UUIDs should be changed. 

The other comments can either be changed in this PR or afterwards, if we want to keep this as just a move.

asbjornsabo committed:
Bluetooth: Audio: CCID - updates for upmerge

Updates to the CCID module for upmerge to Zephyr main

- add bt_ prefix to the ccid_get() function, add explaining comment,
  use defined value for limit
- update copyright statement
- update Kconfig to current practice

Signed-off-by: Asbjørn Sæbø <asbjorn.sabo@nordicsemi.no>
